### PR TITLE
feat: add sign in page (#4)

### LINKED ISSUE #4: Sign-in / Sign-up form
We need a sign-in and sign-up for our dashboard. We authenticate a user by email and password.

### PR DESCRIPTION
**Pull request by JannikWibker**
image
closes #4

### CODE REVIEW

File: app/components/Checkbox.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+const ControlledCheckbox = ({ id, label, checked, disabled, onChange }: CheckboxProps) => {
+  return (
+    <div className="flex justify-center items-center space-x-2">
+      <CheckboxPrimitive.Root
+        onCheckedChange={onChange}
+        checked={checked}
+        disabled={disabled}
+        id={id}
+        className={cx('w-4 h-4 flex items-center mx-[1px] border border-gray-300 rounded focus:outline-none focus:border-indigo-500 focus:border-2 focus:w-[18px] focus:h-[18px] focus:mx-0', {
```
Comment (use-to) [inline app/components/Checkbox.tsx:29]: You should use consistent colors. Look at the borders and the button color. Its different and its weird...

**Review by use-to — CHANGES_REQUESTED** [review]

File: app/components/Checkbox.tsx (hunk 1 above)
Comment (JannikWibker) [reply app/components/Checkbox.tsx:29]: fixed

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

File: app/pages/login.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+              <Checkbox id="login:remember-me" label={translation.stayLoggedIn} onChange={setRememberMe} checked={rememberMe} />
+              <div className="text-sm">
+                <Link href="/forgot-password" passHref>
+                  <a className="font-medium text-indigo-600 hover:text-indigo-500">{translation.forgotPassword}</a>
+                </Link>
+              </div>
+            </div>
+
+            <div className="w-80">
+              <button onClick={handleLogin} className="bg-indigo-600 py-2 px-4 text-sm font-medium text-white rounded-md w-full hover:bg-indigo-700 focus:outline-none focus:ring-indigo-500 focus:ring-2 focus:ring-offset-2">
```
[the remark this answers is not in the document]
Comment (JannikWibker) [reply app/pages/login.tsx]: fixed

**Review by MaxSchaefer — APPROVED** [review]

**Review by use-to — APPROVED** [review]
Looks pretty good to me. We should move the authentication fix to another issue!

File: app/hooks/useAuth.ts [hunk 3]
```diff
@@ -77,22 +77,28 @@ const useAuth = (redirect: () => void, options = defaultUseAuthOptions) => {
...
     }
 
-    // access token expired, use refresh token to acquire new access token and return user data afterwards
+    // access token expired, use refresh token to acquire new access token and set user data afterwards
     if (refreshTokenValid) {
       loginWithRefreshToken(refreshToken)
-        .then(accessToken => {
+        .then(({ accessToken, refreshToken }) => {
+          if (refreshToken !== null) {
+            Cookies.set('jwt-refresh-token', refreshToken)
```
[the remark this answers is not in the document]
Comment (JannikWibker) [reply app/hooks/useAuth.ts:93]: This can be changed later on as it's all mostly mocked anyways and switching to localStorage isn't a great deal of work

**Pull request merged**